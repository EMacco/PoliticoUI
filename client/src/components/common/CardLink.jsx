import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const CardLink = ({ officeId, officeName, numberOfCandidates }) => {
  const link = `/offices?officeId=${officeId}&officeName=${officeName}`;
  return (
    <Fragment>
      Candidates: <Link to={link}>{numberOfCandidates} Candidates</Link>
    </Fragment>
  );
};

CardLink.propTypes = {
  officeId: PropTypes.number.isRequired,
  officeName: PropTypes.string.isRequired,
  numberOfCandidates: PropTypes.number.isRequired
};

export default CardLink;
