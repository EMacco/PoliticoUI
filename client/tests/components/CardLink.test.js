import React from 'react';
import CardLink from '@components/common/CardLink';

const props = {
  officeId: 1,
  officeName: 'Office Name',
  numberOfCandidates: 4
};

const setup = () => {
  const wrapper = shallow(<CardLink {...props} />);
  return wrapper;
};

describe('<CardLink />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setup();
  });

  it('should render the CardLink component correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('CardLink')).toBeTruthy();
  });
});
