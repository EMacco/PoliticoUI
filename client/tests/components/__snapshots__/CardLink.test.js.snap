// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardLink /> should render the CardLink component correctly 1`] = `
<Fragment>
  Candidates: 
  <Link
    to="/offices?officeId=1&officeName=Office Name"
  >
    4
     Candidates
  </Link>
</Fragment>
`;
